lib/grpc/client/adapters/gun.ex

```diff
@@ -1,61 +1,56 @@
 defmodule GRPC.Client.Adapters.Gun do
-  @moduledoc false
+  @moduledoc """
+  A client adapter using Gun
+  """
 
   @behaviour GRPC.Client.Adapter
 
-  # A client adapter using Gun.
-  # conn_pid and stream_ref is stored in `GRPC.Server.Stream`.
+  # conn_pid and stream_ref are stored in `GRPC.Server.Stream`
 
   @default_transport_opts [nodelay: true]
-  @default_http2_opts %{settings_timeout: :infinity}
   @max_retries 100
 
   @impl true
-  def connect(channel, nil), do: connect(channel, %{})
+  def connect(channel, opts \\ %{})
   def connect(%{scheme: "https"} = channel, opts), do: connect_securely(channel, opts)
   def connect(channel, opts), do: connect_insecurely(channel, opts)
 
   defp connect_securely(%{cred: %{ssl: ssl}} = channel, opts) do
-    transport_opts = Map.get(opts, :transport_opts, @default_transport_opts ++ ssl)
-    open_opts = %{transport: :ssl, protocols: [:http2]}
+    transport_opts = Map.get(opts, :transport_opts) || []
 
-    open_opts =
-      if gun_v2?() do
-        Map.put(open_opts, :tls_opts, transport_opts)
-      else
-        Map.put(open_opts, :transport_opts, transport_opts)
-      end
+    tls_opts = Keyword.merge(@default_transport_opts ++ ssl, transport_opts)
```

Comment:
Just quick check from mobile.

which ssl configuration (via cred vs via transport_opts) is higher priority?

Reply:
Via transport options

There's a test that shows that

Reply:
Do we need to document GRPC.Stub.connect about this behaviour? Because the library encourage user to use GRPC.Credential to set ssl credential but now user can set it via adapter option when calling GRPC.Stub.connect.

Reply:
I don't think so. This is a per-adapter specification

Reply:
We can decide to not let some overrides take place, though.

Although I wouldn't invest too much energy in the Gun adapter since we're planning on moving on to Mint

Reply:
So how about 

```elixir
Keyword.merge(@default_transport_opts, transport_opts)
|> Keyword.merge(ssl)
```

i think we should clarify this behaviour before add mint adapter.

Reply:
BTW. Which way we choose, i ok with it. :)

Reply:
I prefer the way it's currently coded, in which the `opts` values have precedence. Could you point to where in the docs it encourages the SSL values to be used? I don't think we'll have to change those other than adding a "opts can override this" warning

Reply:
Here it is https://github.com/elixir-grpc/grpc/blob/master/lib/grpc/credential.ex#L3


Agree that if we can warning the opts can be override.

Reply:
I'll improve on that :)

Reply:
Done :) I'm merging this, but let me know if things are still unclear

Reply:
That’s totally clear :)